diff --git a/config/config-sample.ini b/config/config-sample.ini
index 29c6c60..d2de74e 100644
--- a/config/config-sample.ini
+++ b/config/config-sample.ini
@@ -32,6 +32,10 @@ password = password
 [php_auth]
 enabled = 0
 admin_group = "systems"
+; This option controls whether or not the username is converted  
+; to all lowercase letters. Can be useful if the username is stored
+; in Active Directory (case-insensitive).
+lowercase_user_auth = 0
 
 [ldap]
 enabled = 1
diff --git a/model/user.php b/model/user.php
index d570e3b..d4898fd 100644
--- a/model/user.php
+++ b/model/user.php
@@ -121,7 +121,14 @@ class User extends Record {
 	*/
 	public function get_details_from_php_auth() {
 		global $config;
-		if($this->uid == $_SERVER['PHP_AUTH_USER'] and isset($_SERVER['PHP_AUTH_NAME']) and isset($_SERVER['PHP_AUTH_EMAIL']) and isset($_SERVER['PHP_AUTH_GROUPS'])) {
+		
+		if (!empty($config['php_auth']['lowercase_user_auth'])) {
+			$php_auth_user = mb_strtolower($_SERVER['PHP_AUTH_USER']);
+		} else {
+			$php_auth_user = $_SERVER['PHP_AUTH_USER'];
+		}		
+
+		if($this->uid == $php_auth_user and isset($_SERVER['PHP_AUTH_NAME']) and isset($_SERVER['PHP_AUTH_EMAIL']) and isset($_SERVER['PHP_AUTH_GROUPS'])) {
 			$this->auth_realm = 'PHP_AUTH';
 			$this->name = $_SERVER['PHP_AUTH_NAME'];
 			$this->email = $_SERVER['PHP_AUTH_EMAIL'];
diff --git a/requesthandler.php b/requesthandler.php
index c2107a8..169689f 100644
--- a/requesthandler.php
+++ b/requesthandler.php
@@ -21,7 +21,14 @@ ob_start();
 set_exception_handler('exception_handler');
 
 if(isset($_SERVER['PHP_AUTH_USER'])) {
-	$active_user = $user_dir->get_user_by_uid($_SERVER['PHP_AUTH_USER']);
+
+	if (!empty($config['php_auth']['lowercase_user_auth'])) {
+		$php_auth_user = mb_strtolower($_SERVER['PHP_AUTH_USER']);
+	} else {
+		$php_auth_user = $_SERVER['PHP_AUTH_USER'];
+	}
+
+	$active_user = $user_dir->get_user_by_uid($php_auth_user);
 } else {
 	throw new Exception("Not logged in.");
 }
